### DISCUSSION TITLE
Support external embeds in Quarto Manuscripts of any object created with the new cross-reference Div syntax

### DISCUSSION THREAD

**Discussion by stephan-koenig** [quarto-dev/quarto-cli#7791] [Feature Requests]
### Description

Hi all,

Maybe this is already on the roadmap, but I was wondering if it would be possible in a [Quarto Manuscript to embed](https://quarto.org/docs/manuscripts/authoring/rstudio.html#embed) any object created with the [Cross-Reference Div Syntax](https://quarto.org/docs/authoring/cross-references-divs.html) (instead of only figures and tables based on their code chunk labels)? I believe the Cross-Reference Div Syntax provides more flexibility because it is easy to include, for example, R expressions in the caption, and it would be great to bring that flexibility to embedded figures and tables.

Thanks!

Stephan

**Comment by mcanouil** [comment]
Did you look at the 1.4 change log especially the parts related to cross-reference?

https://quarto.org/docs/download/

Since you can create any custom cross-reference, I believe what you want is already in 1.4.

Could you clarify what you think is missing?

**Reply by cscheid** [reply]
I believe he's talking about [notebook embeds](https://quarto.org/docs/authoring/notebook-embed.html). We haven't hooked that part of the code to the generalized crossref work.

(If I'm correct, then the answer to the OP is: yes, we have thought about it, and yes we'll do it. Just not right now!)

**Reply by stephan-koenig** [reply]
Yes, I was thinking about notebook embeds. Great, thank you for confirming!

**Comment by bschilder** [comment]
> I believe he's talking about [notebook embeds](https://quarto.org/docs/authoring/notebook-embed.html). We haven't hooked that part of the code to the generalized crossref work.
> 
> (If I'm correct, then the answer to the OP is: yes, we have thought about it, and yes we'll do it. Just not right now!)

Could this Discussion be reopened, unless the requested functionality has since been implemented?

**Reply by bschilder** [reply]
It seems it was implemented in Quarto 1.4
https://quarto.org/docs/authoring/cross-references-divs.html

Though I can't seem to get this working for equations. I noticed equation are missing from the docs too, so perhaps this isn't supported for all div content yet?

Doc 1: notebooks/equations.qmd

~~~
::: {#eq-evidence-scores}
```{=tex}
\begin{equation*}
   \frac{1}{2}
\end{equation*} 
```
:::
~~~

Doc 2: index.qmd
~~~

::: {.content-visible unless-format="html"}
{{< embed notebooks/equations.qmd#eq-evidence-scores >}}
:::
~~~

Console output upon rendering Doc 2:

```
==> quarto preview index.qmd --to nature-pdf --no-watch-inputs --no-browse

processing file: equations.qmd
                                                                                                            
output file: equations.knit.md

ERROR: The cell eq-evidence-scores does not exist in notebook

Stack trace:
    at file:///Applications/quarto/bin/quarto.js:69692:23
    at Array.map (<anonymous>)
    at notebookMarkdown (file:///Applications/quarto/bin/quarto.js:69689:40)
    at eventLoopTick (ext:core/01_core.js:183:11)
    at async replaceNotebookPlaceholders (file:///Applications/quarto/bin/quarto.js:69633:32)
    at async renderPandoc (file:///Applications/quarto/bin/quarto.js:69992:28)
    at async Object.onRender (file:///Applications/quarto/bin/quarto.js:93785:33)
    at async renderFileInternal (file:///Applications/quarto/bin/quarto.js:77243:17)
    at async renderFiles (file:///Applications/quarto/bin/quarto.js:77067:17)
    at async renderProject (file:///Applications/quarto/bin/quarto.js:77394:25)
```

**Reply by mcanouil** [reply]
> Could this Discussion be reopened, unless the requested functionality has since been implemented?

Closed by the author.

<img width="284" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/54a35b16-b504-49e5-b678-2ca7a8610a4d">

**Reply by mcanouil** [reply]
@bschilder Previous answer still applies.

> I believe he's talking about [notebook embeds](https://quarto.org/docs/authoring/notebook-embed.html). We haven't hooked that part of the code to the generalized crossref work.
> 
> (If I'm correct, then the answer to the OP is: yes, we have thought about it, and yes we'll do it. Just not right now!)

https://github.com/quarto-dev/quarto-cli/discussions/7791#discussioncomment-7765493

**Reply by bschilder** [reply]
Got it, thanks for reopening and for the quick reply @mcanouil
